core.coffee

```diff
@@ -440,6 +457,57 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				if !mimemessage
```

Comment:
The idea is that you instantiate the client without passing a `mimemessage` dependency in. So if you try to instantiate with `mimemessage` I will check if it has the proper method. In the compileBatch I only need to check if `mimemessage` is present to cover the case when `compiledBatch` is being called without a `mimemessage` dependency passed in. (if it was passed in we would have already checked the method)

Reply:
Ditto about camel case

Reply:
I don't have a strong preference here, I didn't use camel case since the package name is all lower case and I wanted to keep consistent with that choice, e.g. `BluebirdLRU'` is not camel-cased. I am fine to go either way though.